src/game/common/system/localfile.cpp

```diff
@@ -103,227 +103,198 @@ bool LocalFile::Open(const char *filename, int mode)
 {
-    captainslog_trace("Scanning Int from LocalFile %s.", m_filename.Str());
-    char tmp;
-    Utf8String number;
-
     integer = 0;
+    Utf8String number;
+    int count;
+    char tmp;
 
-    // Loop to find the first numeric character.
     do {
-        if (read(m_fileHandle, &tmp, sizeof(tmp)) == 0) {
-            return false;
-        }
-    } while (!isdigit(tmp) && tmp != '-');
+        count = read(m_fileHandle, &tmp, 1);
+    } while (count != 0 && (tmp < '0' || tmp > '9') && tmp != '-');
 
-    // Build up our string of numeric characters
-    while (true) {
-        number.Concat(tmp);
+    if (count == 0) {
+        return false;
+    } else {
+        do {
+            number.Concat(tmp);
+            count = read(m_fileHandle, &tmp, 1);
+        } while (count != 0 && tmp >= '0' && tmp <= '9');
 
-        if (read(m_fileHandle, &tmp, sizeof(tmp)) == 0) {
-            break;
+        if (count != 0) {
+            lseek(m_fileHandle, -1, CURRENT);
         }
 
-        if (!isdigit(tmp)) {
-            // If we read a byte, seek back that byte for the next read
-            // as we are done with the current number.
-            lseek(m_fileHandle, -1, SeekMode::CURRENT);
-
-            break;
-        }
+        integer = atoi(number);
+        return true;
     }
-
-    integer = atoi(number.Str());
-
-    return true;
 }
 
 bool LocalFile::Scan_Real(float &real)
 {
-    captainslog_trace("Scanning Real from LocalFile %s.", m_filename.Str());
-    char tmp;
-    Utf8String number;
-
     real = 0.0f;
+    Utf8String number;
+    int count;
+    char tmp;
+    bool has_point = false;
 
-    // Loop to find the first numeric character.
     do {
-        if (read(m_fileHandle, &tmp, sizeof(tmp)) == 0) {
-            return false;
-        }
-    } while (!isdigit(tmp) && tmp != '-' && tmp != '.');
+        count = read(m_fileHandle, &tmp, 1);
+    } while (count != 0 && (tmp < '0' || tmp > '9') && tmp != '-' && tmp != '.');
 
-    // Build up our string of numeric characters
-    bool have_point = false;
+    if (count == 0) {
+        return false;
+    } else {
+        do {
+            number.Concat(tmp);
 
-    while (true) {
-        number.Concat(tmp);
+            if (tmp == '.') {
+                has_point = true;
+            }
 
-        if (tmp == '.') {
-            have_point = true;
-        }
+            count = read(m_fileHandle, &tmp, 1);
+        } while (count != 0 && tmp >= '0' && tmp <= '9' || tmp == '.' && !has_point);
 
-        if (read(m_fileHandle, &tmp, sizeof(tmp)) == 0) {
-            break;
+        if (count != 0) {
+            lseek(m_fileHandle, -1, CURRENT);
         }
 
-        if (!isdigit(tmp) && (tmp != '.' || have_point)) {
-            // If we read a byte, seek back that byte for the next read
-            // as we are done with the current number.
-            lseek(m_fileHandle, -1, SeekMode::CURRENT);
-
-            break;
-        }
+        real = atof(number);
```

Comment:
number.Str()

Reply:
There is no more need to call .Str here than there is in any of the hundreds of other places across our codebase (including many places elsewhere in the file code) where we rely on Utf8String::operator const char *

If we no longer want to use that operator then we should get rid of it completely and add .Str across the entire codebase but that's for a different PR.